Sources/KlaviyoForms/InAppForms/IAFPresentationManager.swift

```diff
@@ -192,33 +229,59 @@ class IAFPresentationManager {
                 propertiesJSON = "{}"
             }
 
-            let result = try await viewController?.evaluateJavaScript("dispatchProfileEvent('\(event.metric.name.value)', \(propertiesJSON))")
+            let result = try await viewController.evaluateJavaScript("dispatchProfileEvent('\(event.metric.name.value)', \(propertiesJSON))")
             if #available(iOS 14.0, *) {
-                Logger.webViewLogger.info("Successfully dispatched event via Klaviyo.JS\(result != nil ? "; message: \(result.debugDescription)" : "")")
+                Logger.webViewLogger.info("✅ Successfully dispatched event via Klaviyo.JS\(result != nil ? "; message: \(result.debugDescription)" : "")")
             }
         } catch {
             if #available(iOS 14.0, *) {
-                Logger.webViewLogger.warning("Error dispatching event via Klaviyo.JS; message: \(error.localizedDescription)")
+                Logger.webViewLogger.warning("❌ Error dispatching event via Klaviyo.JS; message: \(error.localizedDescription)")
             }
         }
     }
+    
+    /// Replays all pending events that were buffered before webview was ready
+    private func replayPendingEvents() async {
+        guard !pendingEvents.isEmpty else { return }
+        
+        if #available(iOS 14.0, *) {
+            Logger.webViewLogger.info("▶️ Replaying \(self.pendingEvents.count) buffered event(s)")
+        }
+        
+        let eventsToReplay = pendingEvents
+        pendingEvents = []
+        
+        for event in eventsToReplay {
+            try? await handleProfileEventCreated(event)
+        }
+    }
 
     // MARK: - API Key Event Handling
 
     private func reinitializeIAFForNewAPIKey(_ apiKey: String, configuration: InAppFormsConfig) {
```

Comment:
I'm okay getting rid of some of these logs in favor of cleaner code. What do folks think?